proxy/src/telemetry/sensor/http.rs

```diff
@@ -15,6 +15,25 @@ use telemetry::event::{self, Event};
 
 const GRPC_STATUS: &str = "grpc-status";
 
+/// A `RequestOpen` timestamp.
+///
+/// This is added to a request's `Extensions` by the `TimestampRequestOpen`
+/// middleware. It's a newtype in order to distinguish it from other
+/// `Instant`s that may be added as request extensions.
+#[derive(Copy, Clone, Debug)]
+pub struct RequestOpen(pub Instant);
```

Comment:
What does "Open" mean? Is there another term that would better capture this? E.g. "Parsed"?

What is this measuring and what *isn't* it measuring? I think it's important to document that this does not and cannot measure the time the request spend in the TCP receive window and any time spend parsing and/or otherwise pre-processing the request header. Simlarly, we should have some note about what our expectations are on the response end regarding unmeasured overhead. It would be also good to note any other overhead that isn't accounted for.

Reply:
The name "RequestOpen" was chosen because it was previously used here
https://github.com/runconduit/conduit/blob/e9b209829d8b5514b89850bb0b7a64a28b3348b1/proxy/src/telemetry/sensor/http.rs#L215

and elsewhere in the module. I'll improve the documentation.